### DISCUSSION TITLE
Best way to handle analog sweeping across digits

### DISCUSSION THREAD

**Discussion by dwightc** [jomjol/AI-on-the-edge-device#3003] [General] [answered]
I have been using AI on the edge for a few days, and so far it's been accurate most of the time, but my water meter has an annoying design feature: The analog needle sweeps across the entire gauge face. Unfortunately this means that sometimes it obscures one of the digits.

When it completely obscures a digit the built-in error correction features are able to handle it, but I've run into trouble when the needle only partially obscures a digit. This often results in a misread, especially if the needle remains stopped there for a long period of time.

I've looked around and couldn't find anything. Is there an existing way to account for this, such as excluding a digit from consideration if the analog is between certain values?

![alg_roi](https://github.com/jomjol/AI-on-the-edge-device/assets/16245871/682ed3e6-0bcd-4e13-819e-7d3c34d6bf3a)

**Comment by caco3** [comment, marked as answer]
This is tricky to handle, maybe even impossible.

You could try to play with tight max rate and neg rate to try to catch any misreadings.

**Reply by dwightc** [reply]
Thanks for your response. Your suggestion mirrors what I have already done: I have forbidden negative rates, as this meter should never run backwards, and I have set a low max rate to catch implausible reads. Together, these work pretty well. My main remaining challenge is that under high flow rates it can sometimes legitimately outrun my max setting and become stuck if it manages to make several bad reads in a row. So far this has been rare.

If this ends up being a big problem long term, I have figured out how to set up my backend server (Zabbix) to discard any value it receives if the decimal portion is between two set points. However, I want to avoid this if I can.

**Reply by mkmer** [reply]
FWIW, It may also be possible to capture the digits with the needle covering parts of the number and retraining the model to recognize these  partial blockages.
